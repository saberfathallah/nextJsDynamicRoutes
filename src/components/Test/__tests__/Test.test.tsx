import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';

import Test from '../Test';

describe('Test', () => {
  const props = { loading: false, data: { text: 'text' } };
  it('should match snapshout with data', () => {
    const tree = renderer.create(<Test {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('should display text', () => {
    const tree = mount(<Test {...props} />);
    expect(tree.text()).toContain('text');
  });

  it('should match snapshout without data', () => {
    props.loading = true;
    props.data = undefined;
    const tree = renderer.create(<Test {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('should display loading', () => {
    const tree = mount(<Test {...props} />);
    expect(tree.text()).toContain('laoding');
  });
});
