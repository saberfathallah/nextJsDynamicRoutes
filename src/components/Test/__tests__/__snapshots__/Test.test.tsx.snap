// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test should match snapshout with data 1`] = `
<p>
  text
</p>
`;

exports[`Test should match snapshout without data 1`] = `
<p>
  laoding...
</p>
`;
